src/kixi/hecuba/onyx/jobs/weather_fetcher.clj

```diff
@@ -35,31 +60,57 @@
 
 (defn pull-data
   "Get Metoffice data as a string"
-  ([querydate querytime siteid]
-   (->> (client/post "http://datagovuk.cloudapp.net/query"
-                     {:form-params {:Type "Observation"
-                                    :PredictionSiteID siteid
-                                    :ObservationSiteID siteid
-                                    :Date querydate ;; dd/mm/yyyy
-                                    :PredictionTime querytime ;; 0000
-                                    }
-                      :follow-redirects true})
-        :body
-        (re-find #"https://datagovuk.blob.core.windows.net/csv/[a-z0-9]+.csv")
-        slurp
-        )))
+  [querydate querytime siteid]
+  (->> (client/post "http://datagovuk.cloudapp.net/query"
+                    {:form-params {:Type "Observation"
+                                   :PredictionSiteID siteid
+                                   :ObservationSiteID siteid
+                                   :Date querydate ;; dd/mm/yyyy
+                                   :PredictionTime querytime ;; 0000
+                                   }
+                     :follow-redirects true})
+       :body
+       (re-find #"https://datagovuk.blob.core.windows.net/csv/[a-z0-9]+.csv")
+       (client/get)
+       :body
+       ))
 
 (defn pull-weather-station-day-data [querydate siteid]
   (mapv (fn [hour] (try (-> (pull-data querydate (format "%02d00" hour) siteid)
                             (process-data-str)
                             (keep-temp-date-time))
-                        (catch Exception e (str "Exception caught: " (.getMessage e)))))
+                        (catch Exception e (str "Exception caught: " (.getMessage e)))
+                        ))
         (range 0 24)))
 
-;; the result of this message will be serialised and
-;; sent to the outgoing kafka queue (write message)
-;; Onyx expects a map with {:message your-message}
+(defn create-degree-day-measurement [measurements]
+  (let [min-max (get-max-and-min measurements)]
+    {:value (calc-degreedays-mckiver tbase (:min min-max) (:max min-max))
+     :type "Temperature_degreedays"
+     :timestamp (:timestamp (first measurements))}))
+
+(defn create-measurements [measurement-data]
+  (map (fn [reading]
+         (println (str "weather fetcher reading : " reading))
```

Comment:
Logging is out as another story then I can address the whole issue across the three micro services.